### ISSUE TITLE
水果检测例子出错，可能是GPU问题？

### ISSUE THREAD

**Issue by nihuizhidao** [PaddlePaddle/PaddleDetection#494]
你好！

我在一台GTX 1060的笔记本上，CUDA10.0，Ubuntu 16.04系统，跑水果检测的示例代码时，出现了以下错误：

> 2020-04-13 23:52:09,653-INFO: 60 samples in file dataset/fruit/val.txt
> 2020-04-13 23:52:09,653-INFO: places would be ommited when DataLoader is not iterable
> W0413 23:52:10.050293 14612 device_context.cc:237] Please NOTE: device: 0, CUDA Capability: 61, Driver API Version: 10.0, Runtime API Version: 10.0
> W0413 23:52:10.052151 14612 device_context.cc:245] device: 0, cuDNN Version: 7.3.
> W0413 23:52:10.052163 14612 device_context.cc:271] WARNING: device: 0. The installed Paddle is compiled with CUDNN 7.6, but CUDNN version in your machine is 7.3, which may cause serious incompatible bug. Please recompile or reinstall Paddle with compatible CUDNN version.
> 2020-04-13 23:52:10,614-INFO: Found /home/tian/.cache/paddle/weights/yolov3_mobilenet_v1
> 2020-04-13 23:52:10,614-INFO: Loading parameters from /home/tian/.cache/paddle/weights/yolov3_mobilenet_v1...
> 2020-04-13 23:52:10,726-WARNING: /tmp/tmpsn_uyzkw/yolov3_mobilenet_v1.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ]
> 2020-04-13 23:52:10,726-WARNING: /tmp/tmpsn_uyzkw/yolov3_mobilenet_v1.pdparams not found, try to load model file saved with [ save_params, save_persistables, save_vars ]
> 2020-04-13 23:52:10,867-INFO: 240 samples in file dataset/fruit/train.txt
> 2020-04-13 23:52:13,102-INFO: places would be ommited when DataLoader is not iterable
> I0413 23:52:13.274853 14612 parallel_executor.cc:440] The Program will be executed on CUDA using ParallelExecutor, 1 cards are used, so 1 programs are executed in parallel.
> I0413 23:52:13.457108 14612 build_strategy.cc:365] SeqOnlyAllReduceOps:0, num_trainers:1
> I0413 23:52:13.735011 14612 parallel_executor.cc:307] Inplace strategy is enabled, when build_strategy.enable_inplace = True
> I0413 23:52:13.776909 14612 parallel_executor.cc:375] Garbage collection strategy is enabled, when FLAGS_eager_delete_tensor_gb = 0
> /home/tian/.conda/envs/paddle/lib/python3.7/site-packages/paddle/fluid/executor.py:782: UserWarning: The following exception is not an EOF exception.
>   "The following exception is not an EOF exception.")
> Traceback (most recent call last):
>   File "tools/train.py", line 326, in <module>
>     main()
>   File "tools/train.py", line 236, in main
>     outs = exe.run(compiled_train_prog, fetch_list=train_values)
>   File "/home/tian/.conda/envs/paddle/lib/python3.7/site-packages/paddle/fluid/executor.py", line 783, in run
>     six.reraise(*sys.exc_info())
>   File "/home/tian/.conda/envs/paddle/lib/python3.7/site-packages/six.py", line 703, in reraise
>     raise value
>   File "/home/tian/.conda/envs/paddle/lib/python3.7/site-packages/paddle/fluid/executor.py", line 778, in run
>     use_program_cache=use_program_cache)
>   File "/home/tian/.conda/envs/paddle/lib/python3.7/site-packages/paddle/fluid/executor.py", line 843, in _run_impl
>     return_numpy=return_numpy)
>   File "/home/tian/.conda/envs/paddle/lib/python3.7/site-packages/paddle/fluid/executor.py", line 677, in _run_parallel
>     tensors = exe.run(fetch_var_names)._move_to_list()
> paddle.fluid.core_avx.EnforceNotMet: 
> 
> --------------------------------------------
> C++ Call Stacks (More useful to developers):
> --------------------------------------------
> 0   std::string paddle::platform::GetTraceBackString<char const*>(char const*&&, char const*, int)
> 1   paddle::platform::EnforceNotMet::EnforceNotMet(std::__exception_ptr::exception_ptr, char const*, int)
> 2   paddle::operators::BatchNormKernel<paddle::platform::CUDADeviceContext, float>::Compute(paddle::framework::ExecutionContext const&) const
> 3   std::_Function_handler<void (paddle::framework::ExecutionContext const&), paddle::framework::OpKernelRegistrarFunctor<paddle::platform::CUDAPlace, false, 0ul, paddle::operators::BatchNormKernel<paddle::platform::CUDADeviceContext, float>, paddle::operators::BatchNormKernel<paddle::platform::CUDADeviceContext, double>, paddle::operators::BatchNormKernel<paddle::platform::CUDADeviceContext, paddle::platform::float16> >::operator()(char const*, char const*, int) const::{lambda(paddle::framework::ExecutionContext const&)#1}>::_M_invoke(std::_Any_data const&, paddle::framework::ExecutionContext const&)
> 4   paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, paddle::platform::Place const&, paddle::framework::RuntimeContext*) const
> 5   paddle::framework::OperatorWithKernel::RunImpl(paddle::framework::Scope const&, paddle::platform::Place const&) const
> 6   paddle::framework::OperatorBase::Run(paddle::framework::Scope const&, paddle::platform::Place const&)
> 7   paddle::framework::details::ComputationOpHandle::RunImpl()
> 8   paddle::framework::details::FastThreadedSSAGraphExecutor::RunOpSync(paddle::framework::details::OpHandleBase*)
> 9   paddle::framework::details::FastThreadedSSAGraphExecutor::RunOp(paddle::framework::details::OpHandleBase*, std::shared_ptr<paddle::framework::BlockingQueue<unsigned long> > const&, unsigned long*)
> 10  std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, void> >::_M_invoke(std::_Any_data const&)
> 11  std::__future_base::_State_base::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>&, bool&)
> 12  ThreadPool::ThreadPool(unsigned long)::{lambda()#1}::operator()() const
> 
> ------------------------------------------
> Python Call Stacks (More useful to users):
> ------------------------------------------
>   File "/home/tian/.conda/envs/paddle/lib/python3.7/site-packages/paddle/fluid/framework.py", line 2525, in append_op
>     attrs=kwargs.get("attrs", None))
>   File "/home/tian/.conda/envs/paddle/lib/python3.7/site-packages/paddle/fluid/layer_helper.py", line 43, in append_op
>     return self.main_program.current_block().append_op(*args, **kwargs)
>   File "/home/tian/.conda/envs/paddle/lib/python3.7/site-packages/paddle/fluid/layers/nn.py", line 2663, in batch_norm
>     type="batch_norm", inputs=inputs, outputs=outputs, attrs=attrs)
>   File "/home/tian/projects/shennan/PaddleDetection/ppdet/modeling/backbones/mobilenet.py", line 98, in _conv_norm
>     moving_variance_name=bn_name + '_variance')
>   File "/home/tian/projects/shennan/PaddleDetection/ppdet/modeling/backbones/mobilenet.py", line 159, in __call__
>     input, 3, int(32 * scale), 2, 1, name=self.prefix_name + "conv1")
>   File "/home/tian/projects/shennan/PaddleDetection/ppdet/modeling/architectures/yolov3.py", line 61, in build
>     body_feats = self.backbone(im)
>   File "/home/tian/projects/shennan/PaddleDetection/ppdet/modeling/architectures/yolov3.py", line 156, in train
>     return self.build(feed_vars, mode='train')
>   File "tools/train.py", line 117, in main
>     train_fetches = model.train(feed_vars)
>   File "tools/train.py", line 326, in <module>
>     main()
> 
> ----------------------
> Error Message Summary:
> ----------------------
> Error: An error occurred here. There is no accurate error hint for this error yet. We are continuously in the process of increasing hint for this kind of error check. It would be helpful if you could inform us of how this conversion went by opening a github issue. And we will resolve it with high priority.
>   - New issue link: https://github.com/PaddlePaddle/Paddle/issues/new
>   - Recommended issue content: all error stack information
>   [Hint: CUDNN_STATUS_BAD_PARAM] at (/paddle/paddle/fluid/operators/batch_norm_op.cu:287)
>   [operator < batch_norm > error]

我换了一台GTX 1080Ti，CUDA10.0, Ubuntu16.04的却可以正常运行，请问这个是那里出问题了呢？

谢谢！

**Comment by qingqing01** [comment]
@nihuizhidao 可以检查下cuDNN的版本号。 >=1.6.3的Paddle版本需要 cuDNN 7.6.3。参考 https://github.com/PaddlePaddle/Paddle/issues/22510  https://github.com/PaddlePaddle/PaddleDetection/issues/362#issuecomment-601555353

**Comment by nihuizhidao** [comment]
非常感谢！

的确是cuDNN版本不兼容，重新安装7.6版本的cuDNN后就可以了
